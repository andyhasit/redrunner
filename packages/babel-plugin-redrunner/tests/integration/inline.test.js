const {c, EOL, transform, view} = require('../utils')


test('Simple inline', () => {
  const src = view(`
    <div>
      <span>{{name}}</span>
    </div>
  `)
  expect(transform(src)).toMatchSnapshot()
});

test('Inline leaves atts alone', () => {
  const src = view(`
    <div>
      <span>{{name | .getName?}}</span>
      <span id="hey" class="  my-style {{style}}"></span>
    </div>
  `)
  expect(transform(src)).toMatchSnapshot()
});

test('Inline with different endings', () => {
  const src = view(`
    <div>
      <span>{{name|.getName}}</span>
      <span>{{name|.getName?}}</span>
      <span>{{name|.getName()}}</span>
    </div>
  `)
  expect(transform(src)).toMatchSnapshot()
});
