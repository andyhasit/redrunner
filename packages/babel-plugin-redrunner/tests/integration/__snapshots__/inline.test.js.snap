// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline leaves atts alone 1`] = `
"class MyView extends View {}

var __buildUtils__ = MyView.prototype.buildUtils;
MyView.prototype.__ht = '<div><span></span><span id=\\"hey\\" ></span></div>';
MyView.prototype.__wc = [__buildUtils__.getWatch('_1_', undefined, false, {
  'name': function (n, o) {
    this.dom._1_.text(this.getName(n, o));
  }
}), __buildUtils__.getWatch('_2_', undefined, false, {
  'style': function (n, o) {
    this.dom._2_.att('class', \\"my-style \\" + n);
  }
})];
MyView.prototype.queryCollection = __buildUtils__.getQueryCollection({
  'name': function () {
    return this.props.name;
  },
  'style': function () {
    return this.props.style;
  }
});

MyView.prototype.__bv = function (view, prototype) {
  view.__bd(prototype, false);

  view.dom = {
    '_1_': view.__gw([0]),
    '_2_': view.__gw([1])
  };
};"
`;

exports[`Inline with different endings 1`] = `
"class MyView extends View {}

var __buildUtils__ = MyView.prototype.buildUtils;
MyView.prototype.__ht = '<div><span></span><span></span><span></span></div>';
MyView.prototype.__wc = [__buildUtils__.getWatch('_1_', undefined, false, {
  'name': function (n, o) {
    this.dom._1_.text(this.getName);
  }
}), __buildUtils__.getWatch('_2_', undefined, false, {
  'name': function (n, o) {
    this.dom._2_.text(this.getName(n, o));
  }
}), __buildUtils__.getWatch('_3_', undefined, false, {
  'name': function (n, o) {
    this.dom._3_.text(this.getName());
  }
})];
MyView.prototype.queryCollection = __buildUtils__.getQueryCollection({
  'name': function () {
    return this.props.name;
  }
});

MyView.prototype.__bv = function (view, prototype) {
  view.__bd(prototype, false);

  view.dom = {
    '_1_': view.__gw([0]),
    '_2_': view.__gw([1]),
    '_3_': view.__gw([2])
  };
};"
`;

exports[`Simple inline 1`] = `
"class MyView extends View {}

var __buildUtils__ = MyView.prototype.buildUtils;
MyView.prototype.__ht = '<div><span></span></div>';
MyView.prototype.__wc = [__buildUtils__.getWatch('_1_', undefined, false, {
  'name': function (n, o) {
    this.dom._1_.text(n);
  }
})];
MyView.prototype.queryCollection = __buildUtils__.getQueryCollection({
  'name': function () {
    return this.props.name;
  }
});

MyView.prototype.__bv = function (view, prototype) {
  view.__bd(prototype, false);

  view.dom = {
    '_1_': view.__gw([0])
  };
};"
`;
